modules/core/test/test_operations.cpp

```diff
@@ -1561,4 +1561,56 @@ TEST(Core_Arithm, scalar_handling_19599)  // https://github.com/opencv/opencv/is
     EXPECT_EQ(1, c.rows);
 }
 
+TEST(Core_Arith, regression_24163) // https://github.com/opencv/opencv/issues/24163
+{
+    {
+        Mat src1 = ( Mat_<uchar>(1, 16) <<  0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 );
+        Mat src2 = ( Mat_<uchar>(1, 16) <<  1, 1, 1, 1, 1, 1, 1, 1, 1, 1,  1,  1,  1,  1,  1,  1 );
+        Mat mean = ( src1 + src2 ) / 2;
+        Mat expected(1,16,CV_8U);
```

Comment:
Maybe it would be better to use non 8-divisible size to verify that tail processing also works as expected.  E.g. 20 elements.

Comon array size can be stored as a separate constant to reduce duplication.

Reply:
Thank you for your review.
I updated this test including 3x3, 3x4, 3x5, 3x6 cases.

- 3x3 - 9 elements - float32_4 x 2 + float32
- 3x4 - 12 elements - float32_4 x 3
- 3x5 - 15 elements - float32_4 x 3 + float32_2 x 1 + float32 x 1
- 3x6 - 18 elements - float32_4 x 4 + float32_2 x 1

I think these test are including above test.
- 3x1 -  3 elements - float32_2 x 1 + float32 x 1
- 3x2 -  6 elements - float32_4 x 1 + float32x2 x 1

And CV_8S, CV_16U, CV_16S and CV_32S are supported to test.